docs/buffer-mesh.md

```diff
@@ -0,0 +1,170 @@
+                                 	new VertexBufferBinding(colorBuffer, 4)]);
+
+// add vertexElement to tell GPU how to read vertex from vertexBuffer.
+mesh.setVertexElements([new VertexElement("POSITION", 0, VertexElementFormat.Vector3, 0),
+                            new VertexElement("COLOR", 0, VertexElementFormat.NormalizedUByte4, 1)]);
+
+// add one subMesh and set how many vertex you want to render.
+mesh.addSubMesh(0, vertexCount);
+
+// set mesh
+renderer.mesh = mesh;
+```
+
+
+### Instance VertexBuffer
+GPU Instance rendering is a common technology for 3D engines. For example, objects of the same geometric shape can be rendered to different positions at one time, which can greatly improve rendering performance. The main feature of this case is the use of the instance function of [VertexElement](${api}core/VertexElement). The last parameter of the constructor represents the instance step frequency (the number of instances drawn for each vertex in the buffer, non-instance The element must be 0), [instanceCount](${api}core/BufferMesh#instanceCount) in [BufferMesh](${api}core/BufferMesh) represents the number of instances.
+
+```typescript
+// add MeshRenderer component
+const renderer = entity.addComponent(MeshRenderer);
+
+// create mesh
+const mesh = new BufferMesh(engine);
+
+// create vertices.
+const vertices = new ArrayBuffer( vertexByteLength );
+
+// create instance data.
+const instances = new Float32Array( instanceDataLength );
+
+// create vertexBuffer and upload vertex data.
+const vertexBuffer = new Buffer( engine, BufferBindFlag.VertexBuffer, vertices );
+
+// create instance buffer and upload instance data.
+const instanceBuffer = new Buffer( engine, BufferBindFlag.VertexBuffer, instances );
+
+// bind vertexBuffer with stride, stride is every vertex byte length,so the value is 16.
+mesh.setVertexBufferBindings([new VertexBufferBinding( vertexBuffer, 16 ),
+                                  new VertexBufferBinding( instanceBuffer, 12 )]);
+
+// add vertexElement to tell GPU how to read vertex from vertexBuffer.
+mesh.setVertexElements([new VertexElement( "POSITION", 0, VertexElementFormat.Vector3, 0 ),
+                            new VertexElement( "COLOR", 12, VertexElementFormat.NormalizedUByte4, 0 ),
+                            new VertexElement( "INSTANCE_OFFSET", 0, VertexElementFormat.Vector3, 1 , 1 ),
+                            new VertexElement( "INSTANCE_ROTATION", 12, VertexElementFormat.Vector3, 1 , 1 )]]);
+
+// add one sub mesh and set how many vertex you want to render, here is full vertexCount.
+mesh.addSubMesh(0, vertexCount);
+
+// set mesh
+renderer.mesh = mesh;
+```
+
+
+## IndexBuffer
+Using the index buffer can reuse the vertices in the vertex buffer, so as to achieve the purpose of saving video memory. Its usage is very simple, that is, an index buffer object is added on the original basis. The following code is modified on the basis of the first **interleaved vertex buffer** case.
```

Comment:
To achieve the purpose of saving video memory, the vertices in the vertex buffer can be reused.

Reply:
done